elearning-frontend/src/components/admin/Categories.js

```diff
@@ -0,0 +1,272 @@
+import React, { useCallback, useEffect, useRef, useState } from 'react'
+import { useNavigate } from 'react-router-dom'
+import apiClient from '../../services/api'
+
+const Categories = (props) => {
+    
+    const navigate = useNavigate()
+
+    // Cateogry List
+    const [categoryList, setCategoryList] = useState([])
+
+    const [changeData, setChangeData] = useState(true)
+    const [isLoading, setIsLoading] = useState(true)
+
+    // Category
+    const [hasSelectedCategory, setHasSelectedCategory] = useState(false)
+    const [category, setCategory] = useState({
+        id: '',
+        title: '',
+        description: '',
+        slug: '',
+    })
+
+    const usertoken = localStorage.getItem('user')
+
+    // Fetch all categories
+    const FetchCategories = useCallback(async () => {
+        const data = await apiClient({
+            method: "get",
+            url: "/api/admin/get-categories",
+            headers: {
+                Authorization: 'Bearer ' + usertoken
+            }
+        }).then(response => {
+            setCategoryList(response.data.categories)
+            setChangeData(false)
+        }).catch(error => {
+            console.log(error)
+        })
+    }, [changeData])
+
+    // Component Mount
+    useEffect(() => {
+        if (!props.session) {
+            navigate('/')
+        }
+        else {
+            apiClient({
+                method: "get",
+                url: "/api/user",
+                headers: {
+                    Authorization: 'Bearer ' + usertoken
+                }
+            }).then(response => {
+                if (!!response.data.user.profile.is_admin) {
+                    FetchCategories()
+                }
+                else {
+                    navigate('/dashboard')
+                }
+
+                setIsLoading(false)
+            }).catch(error => {
+                console.log(error)
+            })
+        }
+    }, [props.session, FetchCategories, changeData])
+
+    // Save selected category
+    const SaveCategory = (e) => {
+        e.preventDefault()
+
+        if (!hasSelectedCategory && (category.title === '' || category.description === '' || category.slug === '')) {
+            window.alert('No data to save!')
+            return
+        }
+
+        apiClient({
+            method: (hasSelectedCategory ? "put" : "post"),
+            url: (hasSelectedCategory ? "/api/admin/update-category/" + category.id : "/api/admin/add-category"),
+            headers: {
+                Authorization: 'Bearer ' + usertoken,
+                'Content-Type': 'application/json'
+            },
```

Comment:
i guess this was set in your apiClient so dont need to set again, you can remove this one